### PR TITLE
Freezetrack

### PR DESCRIPTION
**Pull request by tomt1664**
Two new in-memory lists added to enable the automatic reissuance (inflation) of assets: assetEntropyMap and freezeHistMap. 

assetEntropyMap stores the entropy and reissuance token for each asset. This list is updated each time a new asset is issued. These values are required to create a raw reissuance transaction (for the original reissueasset RPC, these are found from the issuing wallet, however for a multisig reissuance there is no way to determine them from the asset ID alone). 

freezeHistMap keeps a list of outputs that are tagged as frozen (and the blockheight when they are confirmed) and then records if these outputs are spent (and the spending tx block height). This is required to pause and backdate inflation payments if a redemption is canceled/failed. 

The assetEntropyMap is then used to add the entropy and token information to the asset list output from the getutxoassetinfo RPC. The freezeHistMap is read via a new getfreezehistory RPC. 

Also, a change is VerifyAmounts to allow zero value outputs (so that the freeze-tag output can be zero valued).

### CODE REVIEW

**Comment by tomt1664** [comment]
I removed a check from tests/blind_tests.cpp to allow zero value outputs.

File: src/init.cpp [hunk 1]
```diff
@@ -1108,6 +1109,8 @@ bool AppInitParameterInteraction()
     fAcceptRegisteraddress = GetBoolArg("-registeraddress", DEFAULT_ACCEPT_REGISTERADDRESS);
     nMaxRegisteraddressBytes = GetArg("-registeraddresssize", nMaxRegisteraddressBytes);
 
+    fRecordInflation = GetBoolArg("-recordinflation", DEFAULT_PERMIT_BAREMULTISIG);
```
Comment (nkostoulas) [inline src/init.cpp]: this should be DEFAULT_RECORD_INFLATION

**Comment by tomt1664** [comment]
A redemption tx is marked with a P2PKH output with a null address (unspendable). Before I was sending a dust amount to this output, but it's cleaner to send zero. I don't think there is any issue with allowing zero value outputs - it's set as policy in Bitcoin to prevent UTXO bloat and disincentivise coloured coins etc.

File: src/init.cpp (hunk 1 above)
Comment (tomt1664) [reply src/init.cpp]: fixed

**Pull request merged**